stan/math/prim/scal/err/is_3F2_converges.hpp

```diff
@@ -0,0 +1,71 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_IS_3F2_CONVERGES_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_IS_3F2_CONVERGES_HPP
+
+#include <stan/math/prim/scal/err/is_not_nan.hpp>
+#include <stan/math/prim/scal/fun/is_nonpositive_integer.hpp>
+#include <stan/math/prim/scal/fun/value_of_rec.hpp>
+#include <cmath>
+#include <limits>
+
+namespace stan {
+namespace math {
+
+/**
+ * Check if the hypergeometric function (3F2) called with
+ * supplied arguments will converge, assuming arguments are
+ * finite values.
+ * @tparam T_a1 Type of <code>a1</code>
+ * @tparam T_a2 Type of <code>a2</code>
+ * @tparam T_a3 Type of <code>a3</code>
+ * @tparam T_b1 Type of <code>b1</code>
+ * @tparam T_b2 Type of <code>b2</code>
+ * @tparam T_z Type of <code>z</code>
+ * @param a1 Variable to check
+ * @param a2 Variable to check
+ * @param a3 Variable to check
+ * @param b1 Variable to check
+ * @param b2 Variable to check
+ * @param z Variable to check
+ * @return <code>true</code> if <code>3F2(a1, a2, a3, b1, b2, z)</code>
+ *   meets convergence conditions, and no coefficient is NaN
+ */
+template <typename T_a1, typename T_a2, typename T_a3, typename T_b1,
+          typename T_b2, typename T_z>
+inline bool is_3F2_converges(const T_a1& a1, const T_a2& a2, const T_a3& a3,
+                             const T_b1& b1, const T_b2& b2, const T_z& z) {
+  using std::fabs;
+  using std::floor;
+
+  if (!(is_not_nan(a1) || is_not_nan(a2) || is_not_nan(a3) || is_not_nan(b1)
```

Comment:
Multiple negations should be unwound through the operators.  Unwinding the double negations here, I get 

```
!(!a || !b)  ==  !!a && !!b  ==  a && b
```

and assuming `!is_not_nan(a1) == is_nan(a1)`, we get
```
if (is_nan(a1) && ... && is_nan(z))
  return false;
```

But shouldn't those be `||`?